contribs/gmf/src/directives/print.js

```diff
@@ -0,0 +1,803 @@
+    return this.ngeoPrintUtils_.getOptimalScale(mapSize, viewResolution,
+        this.paperSize_, scales.reverse());
+  }
+  return this.fields.scales[0];
+};
+
+
+/**
+ * @param {!angular.$http.Response} resp Response.
+ * @private
+ */
+gmf.PrintController.prototype.handleCreateReportSuccess_ = function(resp) {
+  var mfResp = /** @type {MapFishPrintReportResponse} */ (resp.data);
+  var ref = mfResp.ref;
+  goog.asserts.assert(ref.length > 0);
+  this.curRef_ = ref;
+  this.getStatus_(ref);
+};
+
+
+/**
+ * @param {string} ref Ref.
+ * @private
+ */
+gmf.PrintController.prototype.getStatus_ = function(ref) {
+  this.requestCanceler_ = this.$q_.defer();
+  this.ngeoPrint_.getStatus(ref, /** @type {angular.$http.Config} */ ({
+    timeout: this.requestCanceler_.promise
+  })).then(
+      this.handleGetStatusSuccess_.bind(this, ref),
+      this.handleCreateReportError_.bind(this)
+  );
+};
+
+
+/**
+ * @param {string} ref Ref.
+ * @param {!angular.$http.Response} resp Response.
+ * @private
+ */
+gmf.PrintController.prototype.handleGetStatusSuccess_ = function(ref, resp) {
+  var mfResp = /** @type {MapFishPrintStatusResponse} */ (resp.data);
+  var done = mfResp.done;
+  if (done) {
+    // The report is ready. Open it by changing the window location.
+    window.location.href = this.ngeoPrint_.getReportUrl(ref);
+    this.resetPrintStates_();
+  } else {
+    // The report is not ready yet. Check again in 1s.
+    var that = this;
+    this.statusTimeoutPromise_ = this.$timeout_(function() {
+      that.getStatus_(ref);
+    }, 1000, false);
+  }
+};
+
+
+/**
+ * @param {!angular.$http.Response} resp Response.
+ * @private
+ */
+gmf.PrintController.prototype.handleCreateReportError_ = function(resp) {
+  this.resetPrintStates_(gmf.PrintState.ERROR_ON_REPORT);
+};
+
+
+/**
+ * @param {number} scale The scale to get the legend (for wms layers only).
+ * @return {Object?} Legend object for print report or null.
+ * @private
+ */
+gmf.PrintController.prototype.getLegend_ = function(scale) {
+  var legend = {'classes': []};
+  var classes, layerNames, layerName, icons;
+  var gettextCatalog_ = this.gettextCatalog_;
+
+  // Get layers from layertree only.
+  var dataLayerGroup = this.ngeoLayerHelper_.getGroupFromMap(this.map,
+      gmf.DATALAYERGROUP_NAME);
+  var layers = this.ngeoLayerHelper_.getFlatLayers(dataLayerGroup);
+
+  // For each visible layer in reverse order, get the legend url.
+  layers.reverse().forEach(function(layer) {
+    classes = [];
+    var source = layer.getSource();
+
+    if (layer.getVisible() && source !== undefined) {
+      // For WMTS layers.
+      if (layer instanceof ol.layer.Tile) {
+        layerName = layer.get('layerName');
+        icons = this.ngeoLayerHelper_.getWMTSLegendURL(layer);
+        // Don't add classes without legend url.
+        if (icons !== null) {
+          classes.push({
+            'name': gettextCatalog_.getString('' + layerName),
```

Comment:
I think that the variable should be `gettextCatalog` to work (only `gettextCatalog.getString` is recognized)